trieste/acquisition/function/multi_objective.py

```diff
@@ -379,3 +384,245 @@ def _update_expected_improvement_fn(
                 isinstance(self._expected_improvement_fn, expected_hv_improvement), []
             )
             self._expected_improvement_fn.update(_partition_bounds)  # type: ignore
+
+
+class HippoPenalizationAcquisitionFunction(GreedyAcquisitionFunctionBuilder[ProbabilisticModel]):
+    r"""
+    HIPPO: HIghly Parallelizable Pareto Optimization
+
+    Builder of the acquisition function for greedily collecting batches by HIPPO
+    penalization in multi-objective optimization. The resulting acquistion function
+    takes in a set of pending points and returns a base acquisition function
+    penalized around those points.
+
+    Penalization is applied to the acquisition function multiplicatively. However, to
+    improve numerical stability, we perform additive penalization in a log space.
+    """
+
+    def __init__(
+        self,
+        objective_tag: str = OBJECTIVE,
+        base_acquisition_function_builder: AcquisitionFunctionBuilder[ProbabilisticModel]
```

Comment:
This needs to be a multi-objective acquisition function. I wonder if there is a way to specify this?

Reply:
In fact, as we only have EHVI, perhaps we should just hard code it and not let the user choose?

Reply:
We also want to support ECHVI, do we not?

Reply:
ExpectedHypervolumeImprovement, BatchMonteCarloExpectedHypervolumeImprovement and ExpectedConstrainedHypervolumeImprovement, right?

t's not something that can currently be inferred automatically. We could explicitly tag them with a dummy MultiObjectiveAcquisitionFunction protocol if we wanted to, and then be able to use that here, but not sure it's worth it.

Reply:
Also if you make this generic, then:

```suggestion
        base_acquisition_function_builder: AcquisitionFunctionBuilder[ProbabilisticModelType]
```

Reply:
Btw, if we don't want ECHVI, the code and usage becomes much easier! But from our discussion of the problems to run HIPPO on i was under the impression that we do

Reply:
Yeah you are right. Perhaps only let us pass one of those two bases?

Reply:
BatchMonteCarloExpectedHypervolumeImprovement we don't want. I agree adding mock interfaces here isn't worth it